core/issuers.js

```diff
@@ -53,21 +62,23 @@ Description must be truthy: ${description}`;
       return mint.mint(assay.empty(), name); // mint and issuer call each other
     },
 
-    getExclusive(amount, srcPaymentP, name = 'a payment') {
-      return Promise.resolve(srcPaymentP).then(srcPayment =>
-        takePayment(amount, false, srcPayment, name),
-      );
+    getExclusive(amount, srcPaymentP, name) {
+      return Promise.resolve(srcPaymentP).then(srcPayment => {
+        name = name !== undefined ? name : srcPayment.getName(); // use old name
+        return takePayment(srcPayment, paymentController, amount, name);
+      });
     },
 
-    getExclusiveAll(srcPaymentP, name = 'a payment') {
-      return Promise.resolve(srcPaymentP).then(srcPayment =>
-        takePayment(
-          mintController.getAmount(srcPayment),
-          false,
+    getExclusiveAll(srcPaymentP, name) {
+      return Promise.resolve(srcPaymentP).then(srcPayment => {
+        name = name !== undefined ? name : srcPayment.getName(); // use old name
```

Comment:
Btw, this is a great example of where the proposed "null coalescing syntax" that just advanced would come in handy. You could correctly rewrite this to

```js
name = name ?? srcPayment.name();
```

which differs only in that it will treat `null` as it treats `undefined`, which is harmless. The way most people (likely including me when I'm not paying enough attention) would code this today is

```js
name = name || srcPayment.getName();
```

which would prevent purposely choosing the empty string as a name. The problem is that the empty string is falsy, which is what `||` tests.

Reply:
Nice! I'm looking forward to being able to use that syntax. Just to make sure, is it ok as is until then?

Reply:
Yes.